Orange/tests/test_metadata_save.py

```diff
@@ -0,0 +1,36 @@
+# Test methods with long descriptive names can omit docstrings
+# pylint: disable=missing-docstring
+
+import io
+from os import path
+import unittest
+import tempfile
+import shutil
+
+from Orange.data import Table, DiscreteVariable
+from Orange.data.io import TabReader
+
+
+class TestMetaData(unittest.TestCase):
+    def setUp(self):
+        DiscreteVariable._clear_cache()
+
+    def test_metadata(self):
+        tempdir = tempfile.mkdtemp()
```

Comment:
This tests assume that iris has metadata and titanic has not. If this would change in the future (if we added metadata for titanic as well), the second test would fail and will need to be fixed, even though the change would have nothing to do with the saving of metadata.

I suggest that you manually set the attributes dict to whatever you need in a test and then do the testing. This way, attributes dict will be the one you expect it to be, even if the datasets change in the future.